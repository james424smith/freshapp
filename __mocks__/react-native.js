import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
