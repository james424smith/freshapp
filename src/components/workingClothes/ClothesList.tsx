import React from "react";
import { View } from "react-native";
import Text from "../StyledText";
import styles from "../../styles/more_screens/workingClothesStyles";
import { ClotheData } from "../../interfaces/workingClothesTypes";
import valueOrDefault from "../../common/valueOrDefault";
import { useTheme, Theme } from "@react-navigation/native";
type Props = { clothesTable: ClotheData[] };

const renderWorkingClothesItem = (
  theme: Theme,
  clotheItem: ClotheData,
  key: number
) => {
  return (
    <View
      key={key}
      style={{
        ...styles.clotheItemView,
        backgroundColor: theme.colors.primary,
      }}
    >
      <Text style={{ ...styles.dateValueSmall, color: theme.colors.text }}>
        {valueOrDefault(clotheItem.issueDate, "")}
      </Text>
      <View>
        <View style={styles.itemContainer}>
          <Text
            style={{ ...styles.detailsValueLarge, color: theme.colors.text }}
          >
            {valueOrDefault(clotheItem.item, "")}
          </Text>
        </View>
        <View style={styles.moreDetailsContainer}>
          <View style={styles.clotheLabels}>
            <Text
              style={{ ...styles.labelsValueBold, color: theme.colors.text }}
            >
              {"Qty:"}
            </Text>
            <Text
              style={{ ...styles.labelsValueBold, color: theme.colors.text }}
            >
              {"Agent:"}
            </Text>
            <Text
              style={{ ...styles.labelsValueBold, color: theme.colors.text }}
            >
              {"Vessel:"}
            </Text>
          </View>
          <View style={styles.moreDetailsValues}>
            <Text
              style={{ ...styles.detailsValueLarge, color: theme.colors.text }}
            >
              {valueOrDefault(clotheItem.quantity, "")}
            </Text>
            <Text
              style={{ ...styles.detailsValueLarge, color: theme.colors.text }}
            >
              {valueOrDefault(clotheItem.issuingAgentName, "")}
            </Text>
            <Text
              style={{ ...styles.detailsValueLarge, color: theme.colors.text }}
            >
              {valueOrDefault(clotheItem.vesselName, "")}
            </Text>
          </View>
        </View>
      </View>
      <View style={styles.lineStyle} />
    </View>
  );
};

const ClotheItemsList = (props: Props) => {
  const { clothesTable } = props;
  const theme = useTheme();
  return (
    <View
      style={{
        ...styles.DataContainerRoot,
        backgroundColor: theme.colors.primary,
      }}
    >
      <View style={styles.clothesContainer}>
        <Text style={{ ...styles.clothesTitle, color: theme.colors.text }}>
          Clothes Allocation
        </Text>
        <View style={styles.clotheDataList}>
          {clothesTable.map((clotheItem: ClotheData, i: number) =>
            renderWorkingClothesItem(theme, clotheItem, i)
          )}
        </View>
      </View>
    </View>
  );
};

export default ClotheItemsList;
