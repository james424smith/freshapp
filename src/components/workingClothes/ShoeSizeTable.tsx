import React from "react";
import { View } from "react-native";
import Text from "../StyledText";
import styles from "../../styles/more_screens/workingClothesStyles";
import { Table, Row, Rows } from "react-native-table-component";
import valueOrDefault from "../../common/valueOrDefault";
import { useTheme } from "@react-navigation/native";

type Props = { headings: string[]; sizeData: (number | undefined)[][] };

const ShoeSizeTable = (props: Props) => {
  const theme = useTheme();
  const { headings, sizeData } = props;
  return (
    <View style={styles.DataContainerRoot}>
      <View
        style={{
          ...styles.shoesContainer,
          backgroundColor: theme.colors.fromBlackToBlue,
        }}
      >
        <View style={styles.tableMarginsStyle}>
          <Text style={styles.shoesTitle}>Shoe Size</Text>
          <View style={styles.tableContainer}>
            <Table
              borderStyle={styles.tableBorderStyle}
              style={{
                ...styles.tableFillStyle,
                backgroundColor: theme.colors.background,
              }}
            >
              <Row
                data={headings}
                style={styles.headingStyle}
                textStyle={{
                  ...styles.dataHeading,
                  color: theme.colors.darkColorOfTable,
                }}
              />
              <Rows
                data={valueOrDefault(sizeData, []) as (number | undefined)[][]}
                textStyle={{
                  ...styles.dataRow,
                  color: theme.colors.darkColorOfTable,
                }}
              />
            </Table>
          </View>
        </View>
      </View>
    </View>
  );
};

export default ShoeSizeTable;
