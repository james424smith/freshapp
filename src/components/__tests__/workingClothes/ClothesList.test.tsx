import React from "react";
import ClothesList from "../../workingClothes/ClothesList";
import { render, cleanup } from "@testing-library/react-native";
import { workingClothes } from "../../../../__mocks__/fake-data/workingClothes.json";

describe("AppColorScheme snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render correct ui and press privacy-policy-button", () => {
    const { toJSON } = render(<ClothesList clothesTable={workingClothes} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
