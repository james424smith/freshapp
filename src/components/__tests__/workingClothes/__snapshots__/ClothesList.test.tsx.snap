// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppColorScheme snapshot test should render correct ui and press privacy-policy-button 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(0, 122, 255)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "paddingTop": 22.5,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2A2E43",
            "fontFamily": "Roboto-Regular",
          },
          Object {
            "color": "rgb(28, 28, 30)",
            "fontSize": 33.5,
            "fontWeight": "700",
            "marginBottom": 45,
            "marginLeft": 45,
            "marginTop": 45,
            "textAlign": "left",
          },
        ]
      }
    >
      Clothes Allocation
    </Text>
    <View
      style={
        Object {
          "paddingTop": 45,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgb(0, 122, 255)",
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 20,
                "marginLeft": 45,
                "textAlign": "left",
              },
            ]
          }
        >
          08-01-2019
        </Text>
        <View>
          <View
            style={
              Object {
                "marginBottom": 45,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 22.5,
                    "fontWeight": "700",
                    "marginBottom": 22.5,
                    "marginHorizontal": 45,
                    "textAlign": "left",
                  },
                ]
              }
            >
              OVERALL
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 22.5,
                      "marginBottom": 22.5,
                      "marginLeft": 45,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Qty:
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 22.5,
                      "marginBottom": 22.5,
                      "marginLeft": 45,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Agent:
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 22.5,
                      "marginBottom": 22.5,
                      "marginLeft": 45,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Vessel:
              </Text>
            </View>
            <View
              style={
                Object {
                  "width": "80%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 22.5,
                      "fontWeight": "700",
                      "marginBottom": 22.5,
                      "marginHorizontal": 45,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                2
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 22.5,
                      "fontWeight": "700",
                      "marginBottom": 22.5,
                      "marginHorizontal": 45,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                MN UKR
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 22.5,
                      "fontWeight": "700",
                      "marginBottom": 22.5,
                      "marginHorizontal": 45,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                LEA AUERBACH
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#78849E",
              "borderBottomWidth": 0.3,
              "marginVertical": 45,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgb(0, 122, 255)",
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 20,
                "marginLeft": 45,
                "textAlign": "left",
              },
            ]
          }
        >
          08-01-2019
        </Text>
        <View>
          <View
            style={
              Object {
                "marginBottom": 45,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 22.5,
                    "fontWeight": "700",
                    "marginBottom": 22.5,
                    "marginHorizontal": 45,
                    "textAlign": "left",
                  },
                ]
              }
            >
              SAFETY SHOES
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "width": "20%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 22.5,
                      "marginBottom": 22.5,
                      "marginLeft": 45,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Qty:
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 22.5,
                      "marginBottom": 22.5,
                      "marginLeft": 45,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Agent:
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 22.5,
                      "marginBottom": 22.5,
                      "marginLeft": 45,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Vessel:
              </Text>
            </View>
            <View
              style={
                Object {
                  "width": "80%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 22.5,
                      "fontWeight": "700",
                      "marginBottom": 22.5,
                      "marginHorizontal": 45,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                1
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 22.5,
                      "fontWeight": "700",
                      "marginBottom": 22.5,
                      "marginHorizontal": 45,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                MN UKR
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2A2E43",
                      "fontFamily": "Roboto-Regular",
                    },
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 22.5,
                      "fontWeight": "700",
                      "marginBottom": 22.5,
                      "marginHorizontal": 45,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                LEA AUERBACH
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#78849E",
              "borderBottomWidth": 0.3,
              "marginVertical": 45,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
